olive/passes/onnx/vitis_ai/quant_utils.py

```diff
@@ -334,21 +359,35 @@ def quantize_data_pof2s(data, qType, symmetric, reduce_range=False, method=Power
     - *S*: scale
     - *z*: zero point
     """
-    assert data.dtype in {
-        np.float16,
-        np.float32,
-        np.dtype("float16"),
-        np.dtype("float32"),
-    }, f"Unexpected dtype {data.dtype!r}"
-    rmin = data.min()
-    rmax = data.max()
-    assert rmin.dtype in {
-        np.float16,
-        np.float32,
-        np.dtype("float16"),
-        np.dtype("float32"),
-    }, f"Unexpected dtype {rmin.dtype!r}"
-    assert rmin.dtype == rmax.dtype
+    if is_ort_version_below_1_17():
+        rmin = 0
+        rmax = 0
+        zero_point = 0
+        scale = 1.0
+        if isinstance(data, np.ndarray):
+            rmin = data.min()
+            rmax = data.max()
+
+        elif isinstance(data, list) and len(data):
+            rmin = min(data)
+            rmax = max(data)
+        data = np.asarray(data)
+    else:
+        assert data.dtype in {
```

Comment:
![image](https://github.com/microsoft/Olive/assets/13343117/3b807343-1808-407f-b20c-19c3c146d220)
![image](https://github.com/microsoft/Olive/assets/13343117/331e9686-eb07-467e-a7c5-8fb74b480614)

Not related with this PR. These types seems a bit duplicated.

Reply:
This is the code Xavier added, I don't touch it.